packages/nimble-components/src/select/specs/IxD.md

```diff
@@ -0,0 +1,254 @@
+
+![ ](spec-images/select-6.png)
+
+#### Icons
+The client-user should be able to specify whether they want left icons on menu items or not. Icons show up in menu items and are displayed as part of the value in the field button. Images and other content not specified here should not be permitted.
+
+![ ](spec-images/select-7.png)
+
+#### Groups
+> QUESTION: IS THE SL "VIEW" PICKER A SELECT (SHOULD IT BE)? WHAT ABOUT GROUPS WITHOUT LABELS, WHAT ABOUT A SELECTION GROUP VS. AN ACTION GROUP OF MENU ITEMS (A.K.A MEGA-MENUS)?
+
+The client-user should be able to create groups for menu items in the menu. Groups are not interactable.
+
+![ ](spec-images/select-8.png)
+
+#### Secondary Text
+> NOTE: NEEDS VISUAL DESIGN
+
+The client-user should be able to specify whether they want secondary text on menu items or not.
+
+![ ](spec-images/select-9.png)
+
+The client-user should be able to specify whether they want other information (including groups) displayed as secondary text in the field input.
+
+![ ](spec-images/select-10.png)
+
+#### Selection
+> NOTE: NOT SURE THIS IS A REAL USE CASE FOR SELECT COMPONENT, SEE FUTURE CONSIDERATIONS FOR MULTI-SELECTION
+
+The client-user should be able to specify whether they want to display selection with or without a check mark.
+
+![ ](spec-images/select-11.png)
+
+#### Filter
+The client-user should be able to specify whether they want filtering available in the select component. The filter field appears in the menu.
+
+![ ](spec-images/select-12.png)
+
+### Behavior
+#### Minimum Width
+> NOTE: NEEDS DEV REVIEW
+
+The field button minimum width is three times the height of the component (to make components targetable on touch screens). The menu width is the same as the field button until reaching the minimum width (without truncating) of the longest string in the menu items.
+
+![ ](spec-images/select-13.png)
+
+#### Menu Height
+The menu height should be enough to display five to seven menu items before overflowing with a vertical scrollbar (not pictured).
+
+#### Overflow Text
+> NOTE: NEEDS DEV & VISUAL DESIGN REVIEW
+
+All text in the select component should truncate, not wrap.
+
+![ ](spec-images/select-14.png)
+
+#### Progress & Loading
+When displaying menu items in the menu lags or is expected to take longer than normal, the menu should indicate that it's loading. The "Loading…" display is not interactive.
+
+![ ](spec-images/select-15.png)
+
+> NOTE: IS THIS NEEDED? IF YES, NEEDS VISUAL DESIGN & TECH WRITER
+
+Menus with dynamically loading menu items may have a "Load more" button. Clicking the "Load more" button should resolve into the "Loading…" indication, then load more menu items and scroll the menu as needed.
+
+![ ](spec-images/select-16.png)
+
+#### Touch-Screen Devices
+On touch-screen devices, the select component should bring up the menu in a Nimble or native tray instead of a popover. 
```

Comment:
While I think it's good to give the proper guidance for mobile devices, this seems like it would be difficult to support. Note that one of the goals for our components was to behave like the native HTML components. Native selects do _not_ have this behavior in browsers on mobile. They open up a popup.

Reply:
You're right, I've updated this.